### PR TITLE
ASoC: rt700/rt711: patches update

### PR DESCRIPTION
**Pull request by shumingfan**
1. ASoC: rt700/rt711: fix pop noise while set the mixer controls
2. ASoC: rt711: use workqueue method to do the calibration

### CODE REVIEW

File: sound/soc/codecs/rt700.c [hunk 1]
```diff
@@ -748,6 +742,60 @@ static int rt700_adc_08_event(struct snd_soc_dapm_widget *w,
 	return 0;
 }
 
+static int rt700_hpo_mux_event(struct snd_soc_dapm_widget *w,
+	struct snd_kcontrol *kcontrol, int event)
+{
+	struct snd_soc_component *component =
+		snd_soc_dapm_to_component(w->dapm);
+	unsigned int val_h = (1 << RT700_DIR_OUT_SFT) | (0x3 << 4);
+	unsigned int val_ll;
```
Comment (plbossart) [inline sound/soc/codecs/rt700.c]: val_h, val_l for symmetry? Not sure what the _ll suffix means?

File: sound/soc/codecs/rt700.c [hunk 2]
```diff
@@ -762,8 +810,12 @@ static const struct snd_soc_dapm_widget rt700_dapm_widgets[] = {
 	SND_SOC_DAPM_DAC_E("DAC Surround", NULL, SND_SOC_NOPM, 0, 0,
 		rt700_dac_surround_event,
 		SND_SOC_DAPM_POST_PMU | SND_SOC_DAPM_PRE_PMD),
-	SND_SOC_DAPM_MUX("HPO Mux", SND_SOC_NOPM, 0, 0, &rt700_hp_mux),
-	SND_SOC_DAPM_PGA("SPK PGA", SND_SOC_NOPM, 0, 0,	NULL, 0),
+	SND_SOC_DAPM_MUX_E("HPO Mux", SND_SOC_NOPM, 0, 0, &rt700_hp_mux,
+		rt700_hpo_mux_event,
+		SND_SOC_DAPM_POST_PMU | SND_SOC_DAPM_PRE_PMD),
+	SND_SOC_DAPM_PGA_E("SPK PGA", SND_SOC_NOPM, 0, 0,	NULL, 0,
```
Comment (plbossart) [inline sound/soc/codecs/rt700.c]: spurious spaces

File: sound/soc/codecs/rt711.c [hunk 3]
```diff
@@ -742,15 +739,29 @@ static int rt711_dac_surround_event(struct snd_soc_dapm_widget *w,
 {
 	struct snd_soc_component *component =
 		snd_soc_dapm_to_component(w->dapm);
+	unsigned int val_h = (1 << RT711_DIR_OUT_SFT) | (0x3 << 4);
+	unsigned int val_ll;
```
Comment (plbossart) [inline sound/soc/codecs/rt711.c]: same, val_l?

File: sound/soc/codecs/rt711-sdw.c [hunk 4]
```diff
@@ -231,6 +231,7 @@ static int rt711_sdw_remove(struct sdw_slave *slave)
 	if (rt711 && rt711->hw_init) {
 		cancel_delayed_work(&rt711->jack_detect_work);
 		cancel_delayed_work(&rt711->jack_btn_check_work);
+		cancel_work_sync(&rt711->cali_work);
```
Comment (plbossart) [inline sound/soc/codecs/rt711-sdw.c]: calibration_work. You have plenty of space on this line :-)

File: sound/soc/codecs/rt711.c [hunk 5]
```diff
@@ -405,10 +405,13 @@ static int rt711_set_jack_detect(struct snd_soc_component *component,
 	struct snd_soc_jack *hs_jack, void *data)
 {
 	struct rt711_priv *rt711 = snd_soc_component_get_drvdata(component);
+	struct snd_soc_dapm_context *dapm =
+		snd_soc_component_get_dapm(rt711->component);
 
 	while (!rt711->hw_init)
 		return -EAGAIN;
 
+	snd_soc_dapm_mutex_lock(dapm);
```
Comment (plbossart) [inline sound/soc/codecs/rt711.c]: for my education, why is this lock needed?

File: sound/soc/codecs/rt711.c [hunk 6]
```diff
@@ -427,6 +430,7 @@ static int rt711_set_jack_detect(struct snd_soc_component *component,
 	rt711->hs_jack = hs_jack;
 
 	dev_dbg(&rt711->slave->dev, "in %s...\n", __func__);
+	snd_soc_dapm_mutex_unlock(dapm);
```
Comment (plbossart) [inline sound/soc/codecs/rt711.c]: you should move this up, maybe after the last regmap_write?

File: sound/soc/codecs/rt711.c (hunk 5 above)
Comment (shumingfan) [reply sound/soc/codecs/rt711.c]: I checked all your comments. Thanks for your suggestions.
I will modify it tomorrow.
For this question, I want to make sure the calibration is done without other factors.
If the machine driver calls snd_soc_component_set_jack API, the calibration work and rt711_set_jack_detect will be interleaved.

Comment (plbossart) [reply sound/soc/codecs/rt711.c]: ok, then a comment would definitively help reviewers if we push this upstream

**Pull request closed without merging**